encoder/v1/encoder.go

```diff
@@ -0,0 +1,155 @@
+package v1
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	encoders "github.com/smartcontractkit/ocr2keepers/encoder"
+	"github.com/smartcontractkit/ocr2keepers/pkg/chain"
+	"github.com/smartcontractkit/ocr2keepers/pkg/types"
+)
+
+const (
+	separator = "|"
+)
+
+var (
+	errUpkeepKeyNotParsable = fmt.Errorf("upkeep key not parsable")
+)
+
+type wrappedPerform struct {
+	CheckBlockNumber uint32   `abi:"checkBlockNumber" json:"checkBlockNumber"`
+	CheckBlockhash   [32]byte `abi:"checkBlockhash" json:"checkBlockhash"`
+	PerformData      []byte   `abi:"performData" json:"performData"`
+}
+
+type reportPacker interface {
+	Pack(args ...interface{}) ([]byte, error)
+	UnpackIntoMap(v map[string]interface{}, data []byte) error
+}
+
+type encoder struct {
+	validator
+	upkeepProvider
+	eligibilityProvider
+	reportKeys []string
+	packer     reportPacker
+}
+
+func NewEncoder() *encoder {
+	arguments := abi.Arguments{
+		{Name: "fastGasWei", Type: chain.Uint256},
+		{Name: "linkNative", Type: chain.Uint256},
+		{Name: "upkeepIds", Type: chain.Uint256Arr},
+		{Name: "wrappedPerformDatas", Type: chain.PerformDataArr},
+	}
+
+	reportKeys := make([]string, len(arguments))
+	for i, arg := range arguments {
+		reportKeys[i] = arg.Name
+	}
+
+	return &encoder{
+		reportKeys: reportKeys,
+		packer:     arguments,
+	}
+}
+
+// EncodeReport encodes a report from a list of upkeep results
+func (e *encoder) EncodeReport(toReport []types.UpkeepResult, _ ...encoders.Config) ([]byte, error) {
+	if len(toReport) == 0 {
+		return nil, nil
+	}
+
+	var baseValuesIdx int
+	for i, rpt := range toReport {
+		// TODO AUTO-2863
+		if rpt.CheckBlockNumber > uint32(baseValuesIdx) {
+			baseValuesIdx = i
+		}
```

Comment:
is it safe to merge with this todo in place?

Reply:
Ah so this was fixed in the PR that was referencing the change that was missed out; this component isn't being used anymore, can you confirm @EasterTheBunny

Reply:
please delete the component if not being used

Reply:
It's going to be used by the OCR3 changes eventually